app/controllers/public/cart_products_controller.rb

```diff
@@ -3,6 +3,14 @@ def index
   end
 
   def create
+      cart_product = CartProduct.new(cart_product_params)
+      cart_product.customer_id = current_customer.id
+    if cart_product.quantity != 0
+      cart_product.save
+      redirect_to cart_products_path
+    else
+      redirect_back(fallback_location: root_path)
+    end
```

Comment:
`:quantity`に1以上の値しか許可しないというのは、`models/cart_product.rb`に、バリデーションを追加してそちらで制御するべきではないでしょうか？
`validates :quantity, :numericality => {:greater_than => 1}`
```suggestion
    cart_product.save
    redirect_to cart_products_path
```